src/diffusers/models/unet_2d.py

```diff
@@ -181,6 +197,7 @@ def forward(
             timesteps = timesteps.reshape((sample.shape[0], *([1] * len(sample.shape[1:]))))
             sample = sample / timesteps
 
-        output = {"sample": sample}
+        if not return_dict:
+            return (sample,)
 
-        return output
+        return UNet2DOutput(sample=sample)
```

Comment:
I'm torn here about whether we should use this other idiom instead:

```Python
        output = UNet2DOutput(sample=sample)
        return output if return_dict else output.to_tuple()
```

- Prevents future mismatch in the tuple vs dict responses, if we add more values in the future and we may forget to update both the tuple and dict paths.
- Shorter
- The `if` goes through the `True` path, which might be easier to parse.

But, is it clearer for readers?

This would be applied to all responses if we decide to do it.

Reply:
Hmm disagree here a bit:
- One liners are harder to read
- We create an unnecessary dict (think some frameworks don't like dicts at all e.g. torchscript)
- We test all outputs so test will fail if there is a mismatch

Reply:
Yeah, curious here if torchscript (and by extension ONNX) would complain about tracing a dict creation.